src/main/java/com/rabbitmq/jms/client/RMQConnection.java

```diff
@@ -82,6 +84,13 @@
     private volatile boolean canSetClientID = true;
 
     /**
+     * QoS setting for channels created by this connection.
+     *
+     * @see com.rabbitmq.client.Channel#basicQos(int)
+     */
+    private final int channelsQos;
```

Comment:
OK, this is not how Java client works but we can safely switch it to per-session (connection) strategy later if JMS users care enough. Going the other way would be a breaking change.